### PR TITLE
Allow for events to be blacklisted

### PR DESCRIPTION
**Pull request by d00rman**
Bug: [T198386](https://phabricator.wikimedia.org/T198386)

### CODE REVIEW

**Comment by Pchelolo** [comment]
LGTM, one nit inline, although adding a test would be neat.

File: lib/base_executor.js [hunk 1]
```diff
@@ -620,6 +644,16 @@ class BaseExecutor {
         }
         return e;
     }
+
+    static compileBlacklist(blacklist) {
```
[the remark this answers is not in the document]
Comment (d00rman) [reply lib/base_executor.js]: {{done}}

**Comment by Pchelolo** [comment]
LGTM. Waiting for travis, merging and deploying.

File: test/feature/update_rules.js [hunk 2]
```diff
@@ -294,6 +323,40 @@ describe('RESTBase update rules', function() {
...
+                'if-unmodified-since': 'Thu, 01 Jan 1970 00:00:01 +0000',
+                'user-agent': 'SampleChangePropInstance'
+            }
+        })
+        .get('/api/rest_v1/page/html/User%3ANolelover/1234')
+        .query({ redirect: false })
+        .reply(200, { });
+
+        return P.try(() => producer.produce('test_dc.mediawiki.revision-create', 0,
+            Buffer.from(JSON.stringify({
```
Comment (Pchelolo) [inline test/feature/update_rules.js]: Mental note: eventually move the creation of these common events into common utilities.

**Comment by Pchelolo** [comment]
New tests are failing!

**Pull request merged**